### ISSUE TITLE
[Tag list] Clarification for interaction for assistive technology users

### ISSUE THREAD

**Issue by Magsu** [sl-design-system/components#2869]
### Provide a general summary of the issue here

Currently the tag list is a list. The element has an interaction, where when focused, users can move between tag items using arrow keys. This is a great idea, especially when there are many items in the tag list, so users have only one tab stop, but the list remains interactive. However, the interaction isn't that clear to assistive technology users. 

The list role is not interactive semantically (this is the reason, why for example NVDA users must set focus to the list items using NVDA key + Space, rather than having the list focusable and interactable when entering in focus mode). List can have interactable elements inside the list (e.g., buttons: #2868), but the list itself typically is just a container, with the added semantics of the items being in a list, and therefore providing additional information of their position in the list. In this case the list contains the the interactable items in one tab stop, so some users might think the items in the list are broken, as the rest cannot be focused using tab. 

Fixing issue #2868 does help with this somewhat, because the button role helps users understand that the items are interactable, but the interaction isn't obvious from the roles.

### 🤔 Expected Behavior?

Users have a clear understanding of the list actions.

### 😯 Current Behavior

Users must use trial and error to understand the interaction.

### 💁 Possible Solution

Some form of clarification for the list interaction should help. Adding a clarifying text with aria-describedby would suffice. E.g., "Use arrow keys to move between list items when focused". This would also be read to users using assistive technology without a keyboard, might need some thinking in how to present it to users.

This element acts basically like a [Layout Grid (W3C ARIA Authoring Practices)](https://www.w3.org/WAI/ARIA/apg/patterns/grid/examples/layout-grids/). It could be refactored as one to make it more obvious, that the interaction is present. **NOTE:** The grid role expects that all the items in the grid are interactive, so this doesn't work with the Mixed variant

### 🖥️ Steps to Reproduce

**NVDA**
1. Press arrow keys to move through the list -> reads list item labels
2. Enter focus mode (NVDA key + Space) -> Shows focus, but still uses NVDA interactions with arrow keys
3. Set focus mode again (NVDA key + Space) -> Actually focuses the buttons and can switch the items in the list using arrow keys

### What browsers are you seeing the problem on? (only for bugs in code)

Chrome

### What operating system are you using?

Windows 11

### 👤 Your name

Valtteri Järvinen

### 🧢 Your product/team

External

### 🎨 Your Theme(s)

Sanoma Learning

**Comment by arecuenco-dsgn** [comment]
After reviewing this and the current component, I think the changes are mainly visuals to make it clear that in removable tags, the interactive part is the **remove button**, not the whole tag.
 
- Only the removable tags can be disabled.
- You can focus the whole tag when they are disabled.

### Visual changes

- Keep the label area neutral and informational as it is.
- Keep the main interaction feedback to the remove button: hover and pressed.
- Changes the focus to the remove button (Inner Focus Ring), instead of focusing on the whole tag. 
- Keep the remove button visible on disabled tags. Disabled only the remove button, not the entire tag.
- Are we sure that the remove button has enough space to be the only actionable element?
- The list should contain only tags that are all static or all removable, but not both in the same list.

### Documentation changes

- Update the examples on the Tag page so they no longer show mixed lists. Show both tag lists (static and removable)
- Adjust the usage content to reflect that removable tags are for editable/removable content, and that mixed lists should not be used.
- Update the anatomy/content so it is clear that the remove button is the only interactive part of a removable tag.
